libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -274,6 +275,8 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_BatteryState& msg, const uint8_
     }
 
     update_topic(msg.header.stamp);
+    std::sprintf(msg.header.frame_id, "%d", instance);
```

Comment:
You can't use `std::sprintf`. Use `snprintf`. It will need a length argument, which you could set to 2 and to be sure, add a static assert the number of battery monitors is less than 99.